lunavl/sdk/indexes/base.py

```diff
@@ -0,0 +1,145 @@
+
+    def __init__(self, coreIndex: Union[IIndexBuilderPtr, IDenseIndexPtr, IDynamicIndexPtr], faceEngine: PyIFaceEngine):
+        """
+        Init index.
+
+        Args:
+            coreIndex: core index class
+            faceEngine (PyIFaceEngine): core face engine
+        """
+        self._coreIndex = coreIndex
+        self._faceEngine = faceEngine
+
+    @property
+    def bufSize(self) -> int:
+        """Get storage size with descriptors."""
+        return self._coreIndex.size()
+
+    @property
+    def descriptorVersion(self) -> int:
+        """Get descriptor version from index."""
+        return self._coreIndex.getDescriptorVersion()
+
+    def __getitem__(self, index: int) -> FaceDescriptor:
+        """
+        Get descriptor by index from internal storage.
+        Args:
+            index: identification of descriptors position in internal storage
+        Raises:
+            IndexError: if index out of range
+            LunaSDKException: if an error occurs while getting descriptor
+        Returns:
+            descriptor
+        """
+        if index >= self.bufSize:
+            # todo remove after fix FSDK-2897 index error
+            raise IndexError(f"Descriptor index '{index}' out of range")
+
+        _coreDescriptor = self._faceEngine.createDescriptor(self._coreIndex.getDescriptorVersion())
+        error, descriptor = self._coreIndex.descriptorByIndex(index, _coreDescriptor)
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        return FaceDescriptor(descriptor)
+
+    def __delitem__(self, index: int) -> None:
+        """
+        Descriptor will be removed from the graph (not from the internal storage), so it is not available for search.
+        Args:
+            index: identification of descriptors position in internal storage
+        Raises:
+            IndexError: if index out of range
+            LunaSDKException: if an error occurs while remove descriptor failed
+        """
+        if index >= self.bufSize:
+            # todo remove after fix FSDK-2897 index error
+            raise IndexError(f"Descriptor index '{index}' out of range")
+
+        error = self._coreIndex.removeDescriptor(index)
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+
+    def checkDescriptorVersion(self, receivedDescriptorVersion: int) -> None:
+        """
+        Check descriptor version with version in index.
+        todo: remove this function after normalizing errors (FSDK-2897)
+        Args:
+            receivedDescriptorVersion: received descriptor version
+        Raises:
+            LunaSDKException(InvalidDescriptor): if descriptor versions do not match
+        """
+        if receivedDescriptorVersion != self.descriptorVersion:
+            raise LunaSDKException(
+                LunaVLError.InvalidDescriptor.format(f"Not expected descriptor version {receivedDescriptorVersion}")
```

Comment:
may be smth like `Expected descriptor version 1, got version 2`?

Reply:
fixed